spring-integration-smb/src/test/java/org/springframework/integration/smb/SmbTestSupport.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.smb;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+
+import org.springframework.context.support.GenericXmlApplicationContext;
+import org.springframework.integration.file.remote.RemoteFileTestSupport;
+import org.springframework.integration.file.remote.session.CachingSessionFactory;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.smb.session.SmbSessionFactory;
+
+import jcifs.DialectVersion;
+import jcifs.smb.SmbFile;
+
+/**
+ * Provides a connection to an external SMB Server for test cases.
+ *
+ * The actual SMB share must be configured in file 'SmbTests-context.xml'
+ * with the 'real' server settings for testing.
+ *
+ * @author Gregory Bragg
+ *
+ * @since 6.0
+ */
+
+@Disabled("Actual SMB share must be configured in file [SmbTests-context.xml].")
```

Comment:
i wonder if we can look into using a Testcontainers with Samba Docker image for this kind of tests: https://hub.docker.com/r/dperson/samba ...

Not the current PR though

Reply:
I definitely looked into some sort of embedded SMB Server for a few days, like what we have for (S)FTP however there is none available. I agree we could have Samba container for the build process to use instead, it doesn't help however for localhost testing against an external SMB share, unless we expose it to the outside world to use for development purposes.

Reply:
No, my point is to to use Testcontainers and its ability to run any Docker image.

That's must not be a part of a build process: just a Unit test scope.
Not sure why you talk about outside world...